### PR TITLE
docs(nx): add an intro tutorial

### PR DESCRIPTION
**Pull request by vsavkin**
Every step of the tutorial has three sections divided by '-----':

- content
- question
- answers (the first answer is always correct)

### CODE REVIEW

File: docs/tutorial/06-proxy.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# Step 6: Proxy
```
Comment (FrozenPandaz) [inline docs/tutorial/06-proxy.md]: I feel like we shouldn't have a page for this.

This has no action for the user to take and is tangent to the process

The previous page should have a note pointing out they pass in `frontendProject` to setup the frontend project to work with the node application.

File: docs/tutorial/02-add-e2e-test.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Step 2: Add E2E Test
```
Comment (FrozenPandaz) [inline docs/tutorial/02-add-e2e-test.md]: I think we should add here that, we're making an E2E test which will test the entire stack of the application. It fails for now, but we will learn how to do both the frontend and backend work all in 1 place to make it pass.

File: docs/tutorial/07-share-code.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+└── tslint.json
+```
+
+**Copy the interface into the library's index file.**
+
+```typescript
+interface Todo { title: string }
+```
+
+## Update Backend
```
Comment (FrozenPandaz) [inline docs/tutorial/07-share-code.md]: Perhaps, this should be a separate step.

File: docs/tutorial/07-share-code.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+
+  addTodo() {
+    this.todos.push({
+      title: `New todo ${Math.floor(Math.random() * 1000)}`
+    });
+  }
+}
+```
+
+## Update Frontend
```
Comment (FrozenPandaz) [inline docs/tutorial/07-share-code.md]: Perhaps, this should be a separate step or alongside the above separate step.

File: docs/tutorial/08-create-libs.md [hunk 5]
```diff
@@ -0,0 +1,161 @@
+# Step 8: Create Libs
```
Comment (FrozenPandaz) [inline docs/tutorial/08-create-libs.md]: This feels out of place... what do you think about making this Step 3 where a Todo component is created in a lib?

File: docs/tutorial/06-proxy.md (hunk 1 above)
Comment (vsavkin) [reply docs/tutorial/06-proxy.md]: I actually had it as one step originally, but I felt like "Creating a node app" and "Connect frontend to Node"  was easier to follow. So I wanted to make sure they run serve for backend first.

File: docs/tutorial/08-create-libs.md (hunk 5 above)
Comment (vsavkin) [reply docs/tutorial/08-create-libs.md]: Yeah, it probably should be :) I tried to match the video I recorded, so folks can watch the video and the do the tutorial. Reordering the steps will require changing the video :) So I'd do those in sync.

File: docs/tutorial/02-add-e2e-test.md (hunk 2 above)
Comment (vsavkin) [reply docs/tutorial/02-add-e2e-test.md]: Fixed

**Pull request merged**